python/ccxt/base/exchange.py

```diff
@@ -1588,6 +1577,12 @@ def find_timeframe(self, timeframe, timeframes=None):
                 return key
         return None
 
+    def clone(self, obj):
+        return obj if isinstance(obj, list) else self.extend(obj)
+
+    def removeSubKeyFromInstance(self, parentKey, subKey):
```

Comment:
do we need to be so specific here? subKey from instance, can't we remove any value from any object using the key?

Reply:
I've pushed last commit now - did you mean that change?

Reply:
@ttodua almost, but in that case `fromInstance` does not make sense anymore right? So we can probably rename it.

Reply:
@ttodua can we rename it to something like `removeValueFromObject` and instead of providing the subKey and parent key we provide them directly so instead of `removeSubKeyFromInstance(this, 'urls', 'api')` we would do something like `removeValueFromKey(this.urls, 'api')` ?

Reply:
I wanted to refrain doing that. the reason is that objects are passed by reference as you know, and the `this.urls` is array in php, which will be passed as a copy (instead of reference), that's why I wanted specifically to have `removeSubKeyFromInstance`, because it would directly affect the target "reference object":

```php
function myFunc(passedObject, ....) {
    passedObject->urls['api'] = whatever;
}

myFunc(this, ...);
```
as opposed to
```php
function myFunc(passedArray, ....) {
    passedArray['api'] = whatever;
}

myFunc(obj->urls, ); // here the value is array, which will be passed to function
```

Reply:
but wait, maybe I am wrong, I will re-test now to ensure.

Reply:
yeah, it's that way, you can confirm here :  https://onecompiler.com/php/3zc5xrwzw

Reply:
@ttodua I understand, but still the current name is confusing, what is "removing a subkey from an instance" ? In reality we're updating an array that is an instance prop

Reply:
the name was not well thought by me, named it on the fly. we can name it `removeItemFromInstanceProp`. is that ok?

Reply:
@ttodua what if we do something like `updateInstanceProp(prop, newValue)` and we provided it with the prop name and the array already updated?

Reply:
in the code we were removing some items from dict/array, so "removing" item needs to be handled differently (unset/delete ...), than "setting" a value to some prop.

Reply:
@ttodua I see, but what if we split it into two different steps? add `deleteElementFromObject` and `updateInstanceProp` and we would do:

```Typescript
const newObject = this.deleteElementFromObject(this.urls, 'api');
this.updateInstanceProp('urls', newObject)
```

⬆️ doing this seems a bit more clear and makes the functions more generic to be used elsewhere

Reply:
it would entail adding two base methods and havent intended to overcomplicate, but sure, I'll go and add that

Reply:
uh oh. so,that approach will fix php's caveats and will work as expected in php, but now it messes up js.
the original object is being mutated that way. to avoid that, we have to additionally clone the object inside `deleteElementFromObject`.

https://jsfiddle.net/kcuymp9q/

should we clone? or use one-liner dedicated method (i.e. `removeDictionaryKeyFromObject`) for such cases that works same way in all langs.

Reply:
ok, i have added an additional clone to avoid mutation, and removed the old oneliner method completely.
lmk once you review.

Reply:
@ttodua sounds good, will test it